kiagnose/internal/launcher/launcher_test.go

```
@@ -122,9 +157,24 @@ func (s checkupStub) Teardown() error {
 }
 
 type reporterStub struct {
-	failReport error
+	reportErr   error
+	reportCount int
 }
 
-func (r reporterStub) Report() error {
-	return r.failReport
+func (r *reporterStub) Report(_ status.Status) error {
+	r.reportCount++
+	if r.reportCount > 2 {
+		panic("Report was called more than twice")
+	}
+
+	if r.reportErr == nil {
+		return nil
+	}
+
+	if r.reportCount == 1 && r.reportErr == errorFailOnInitialReport ||
+		r.reportCount == 2 && r.reportErr == errorFailOnFinalReport {
+		return r.reportErr
+	}
```

Can this be simplified to: `return r.reportErr` ?
Maybe you do not even need to counter now.

If it returns the wrong error, you will catch it.

---

I think we need to return an error only when it is required.
For example, is the error should be on the second report, IMO we should return `nil`, and not an error.

---

Right.

---

Hmm.. but then, why do you need the `	if r.reportErr == nil` condition? Isn't it redundant?

---

It was a left over and was removed.
Thanks.